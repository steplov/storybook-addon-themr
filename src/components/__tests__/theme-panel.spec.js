import React from 'react';
import { shallow } from 'enzyme';
import ThemePanel from '../theme-panel';

const themes = {
  a: {
    a: 'a',
    b: 'b'
  },
  c: {
    d: 'd',
    e: 'e'
  }
};

describe('<ThemePanel />', () => {
  it('should render empty without props', () => {
    const wrapper = shallow(<ThemePanel />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders themes', () => {
    const wrapper = shallow(
      <ThemePanel themes={themes} />
    );
    const options = wrapper.find('option');

    expect(options.get(0).props.value).toBe('a');
    expect(options.get(0).props.children).toBe('a');
    expect(options.get(1).props.value).toBe('c');
    expect(options.get(1).props.children).toBe('c');
  });

  it('renders selected theme', () => {
    const wrapper = shallow(
      <ThemePanel
        themes={themes}
        currentTheme="c"
      />
    );
    const select = wrapper.find('select');

    expect(select.props().value).toBe('c');
  });

  it('handles theme change', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <ThemePanel
        themes={themes}
        onThemeChange={onChange}
      />
    );
    const select = wrapper.find('select');

    select.simulate('change', { target: { value: 'c' } });

    expect(onChange).toHaveBeenCalledWith('c');
  });
});
