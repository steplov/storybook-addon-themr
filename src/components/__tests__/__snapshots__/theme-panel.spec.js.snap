// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemePanel /> should render empty without props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={
        Object {
            "alignContent": "stretch",
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "fontFamily": "
              -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
              \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
            ",
            "justifyContent": "center",
            "padding": 16,
          }
    }
>
    <div
        style={
            Object {
                "marginRight": 16,
                "minWidth": 160,
              }
        }
    >
        <select
            onChange={[Function]}
            style={
                Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "border": "1px solid transparent",
                    "borderColor": "#dbdbdb",
                    "borderRadius": "3px",
                    "boxShadow": "none",
                    "color": "#363636",
                    "cursor": "pointer",
                    "display": "inline-flex",
                    "fontSize": "0.875em",
                    "height": "2.25em",
                    "justifyContent": "flex-start",
                    "lineHeight": "1.5",
                    "maxWidth": "100%",
                    "outline": "none",
                    "paddingBottom": "calc(0.375em - 1px)",
                    "paddingLeft": "calc(0.625em - 1px)",
                    "paddingRight": "calc(0.625em - 1px)",
                    "paddingTop": "calc(0.375em - 1px)",
                    "position": "relative",
                    "verticalAlign": "top",
                    "width": "100%",
                  }
            }
            value=""
        >
            
        </select>
    </div>
    <div
        style={
            Object {
                "fontSize": "0.875em",
                "maxHeight": "80%",
                "overflow": "scroll",
              }
        }
    />
</div>,
  "nodes": Array [
    <div
      style={
            Object {
                  "alignContent": "stretch",
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "flexWrap": "nowrap",
                  "fontFamily": "
                  -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
                  \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
                ",
                  "justifyContent": "center",
                  "padding": 16,
                }
      }
>
      <div
            style={
                  Object {
                        "marginRight": 16,
                        "minWidth": 160,
                      }
            }
      >
            <select
                  onChange={[Function]}
                  style={
                        Object {
                              "alignItems": "center",
                              "backgroundColor": "white",
                              "border": "1px solid transparent",
                              "borderColor": "#dbdbdb",
                              "borderRadius": "3px",
                              "boxShadow": "none",
                              "color": "#363636",
                              "cursor": "pointer",
                              "display": "inline-flex",
                              "fontSize": "0.875em",
                              "height": "2.25em",
                              "justifyContent": "flex-start",
                              "lineHeight": "1.5",
                              "maxWidth": "100%",
                              "outline": "none",
                              "paddingBottom": "calc(0.375em - 1px)",
                              "paddingLeft": "calc(0.625em - 1px)",
                              "paddingRight": "calc(0.625em - 1px)",
                              "paddingTop": "calc(0.375em - 1px)",
                              "position": "relative",
                              "verticalAlign": "top",
                              "width": "100%",
                            }
                  }
                  value=""
            >
                  
            </select>
      </div>
      <div
            style={
                  Object {
                        "fontSize": "0.875em",
                        "maxHeight": "80%",
                        "overflow": "scroll",
                      }
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ThemePanel
        currentTheme=""
        onThemeChange={[Function]}
        themes={Object {}}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ThemePanel {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "props": Object {
          "currentTheme": "",
          "onThemeChange": [Function],
          "themes": Object {},
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={
                    Object {
                              "alignContent": "stretch",
                              "alignItems": "center",
                              "display": "flex",
                              "flexDirection": "row",
                              "flexWrap": "nowrap",
                              "fontFamily": "
                          -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
                          \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
                        ",
                              "justifyContent": "center",
                              "padding": 16,
                            }
          }
>
          <div
                    style={
                              Object {
                                        "marginRight": 16,
                                        "minWidth": 160,
                                      }
                    }
          >
                    <select
                              onChange={[Function]}
                              style={
                                        Object {
                                                  "alignItems": "center",
                                                  "backgroundColor": "white",
                                                  "border": "1px solid transparent",
                                                  "borderColor": "#dbdbdb",
                                                  "borderRadius": "3px",
                                                  "boxShadow": "none",
                                                  "color": "#363636",
                                                  "cursor": "pointer",
                                                  "display": "inline-flex",
                                                  "fontSize": "0.875em",
                                                  "height": "2.25em",
                                                  "justifyContent": "flex-start",
                                                  "lineHeight": "1.5",
                                                  "maxWidth": "100%",
                                                  "outline": "none",
                                                  "paddingBottom": "calc(0.375em - 1px)",
                                                  "paddingLeft": "calc(0.625em - 1px)",
                                                  "paddingRight": "calc(0.625em - 1px)",
                                                  "paddingTop": "calc(0.375em - 1px)",
                                                  "position": "relative",
                                                  "verticalAlign": "top",
                                                  "width": "100%",
                                                }
                              }
                              value=""
                    >
                              
                    </select>
          </div>
          <div
                    style={
                              Object {
                                        "fontSize": "0.875em",
                                        "maxHeight": "80%",
                                        "overflow": "scroll",
                                      }
                    }
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={
                    Object {
                              "alignContent": "stretch",
                              "alignItems": "center",
                              "display": "flex",
                              "flexDirection": "row",
                              "flexWrap": "nowrap",
                              "fontFamily": "
                          -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
                          \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
                        ",
                              "justifyContent": "center",
                              "padding": 16,
                            }
          }
>
          <div
                    style={
                              Object {
                                        "marginRight": 16,
                                        "minWidth": 160,
                                      }
                    }
          >
                    <select
                              onChange={[Function]}
                              style={
                                        Object {
                                                  "alignItems": "center",
                                                  "backgroundColor": "white",
                                                  "border": "1px solid transparent",
                                                  "borderColor": "#dbdbdb",
                                                  "borderRadius": "3px",
                                                  "boxShadow": "none",
                                                  "color": "#363636",
                                                  "cursor": "pointer",
                                                  "display": "inline-flex",
                                                  "fontSize": "0.875em",
                                                  "height": "2.25em",
                                                  "justifyContent": "flex-start",
                                                  "lineHeight": "1.5",
                                                  "maxWidth": "100%",
                                                  "outline": "none",
                                                  "paddingBottom": "calc(0.375em - 1px)",
                                                  "paddingLeft": "calc(0.625em - 1px)",
                                                  "paddingRight": "calc(0.625em - 1px)",
                                                  "paddingTop": "calc(0.375em - 1px)",
                                                  "position": "relative",
                                                  "verticalAlign": "top",
                                                  "width": "100%",
                                                }
                              }
                              value=""
                    >
                              
                    </select>
          </div>
          <div
                    style={
                              Object {
                                        "fontSize": "0.875em",
                                        "maxHeight": "80%",
                                        "overflow": "scroll",
                                      }
                    }
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ThemePanel
    currentTheme=""
    onThemeChange={[Function]}
    themes={Object {}}
/>,
}
`;
